core/src/word_lock.rs

```diff
@@ -52,11 +55,23 @@ fn with_thread_data<F, T>(f: F) -> T
 where
     F: FnOnce(&ThreadData) -> T,
 {
+    // REVIEW: this seems like it could be simplified:
+    //
+    //  if cheap {
+    //      f(&ThreadData::new())
+    //  } else {
+    //      THREAD_DATA.with(f)
+    //  }
+    //
+    // and that's it?
```

Comment:
In an old PR I changed `with_thread_data` to call `f` in two places, a bit like above. But @Amanieu argued that it shuold only be called once, so it's not inlined twice into `with_thread_data`. See this comment: https://github.com/Amanieu/parking_lot/pull/108#discussion_r238117165

Reply:
I think I would personally prefer to err on the side of safety rather than code size for now and "hope llvm takes care of it", and we can evaluate code size impact and such later. It's unlikely to have any real effect I'd imagine since there's just so much code in general in Rust to micro-optimize any one case.